website/static/js/registrationUtils.js

```diff
@@ -741,34 +738,33 @@ RegistrationEditor.prototype.save = function() {
     var data = {};
     $.each(schema.pages, function(i, page) {
         $.each(page.questions, function(qid, question) {
-            if(question.type === 'object'){
+            if (question.type === 'object') {
                 var value = {};
                 $.each(question.properties, function(prop, subQuestion) {
                     value[prop] = {
-                        value: subQuestion.value()
+                        value: subQuestion.value(),
+                        comments: ko.toJS(subQuestion.comments())
                     };
                 });
                 data[qid] = value;
-            }
-            else {
+            } else {
                 data[qid] = {
-                    value: question.value()
+                    value: question.value(),
+                    comments: ko.toJS(question.comments())
                 };
             }
         });
     });
 
-    if (typeof self.draft().pk === 'undefined'){
+    if (typeof self.draft().pk === 'undefined') {
```

Comment:
What is the advantage of this over `if (self.draft().pk`? Is there a situation where pk is falsy and the more specific check is necessary?